lib/msf/core/payload/stager.rb

```diff
@@ -156,7 +156,14 @@ def handle_connection(conn, opts={})
     # If the stage should be sent over the client connection that is
     # established (which is the default), then go ahead and transmit it.
     if (stage_over_connection?)
-      p = generate_stage
+      uuid_raw = conn.get_once(16, 1)
```

Comment:
Handlers are tied to payloads explicitly now already, so no real loss. The idea with universal listeners is to bridge that once all of the UUID stagers are ready. If the stager doesn't send a UUID and we expect one, there is something seriously wrong, so a timeout between 10-30s is reasonable, however that might block up the listener queue. We have some existing stuff in place to help with that, but can address that in future PRs if its an issue.